### DISCUSSION TITLE
Reorder with a collapse item inside

### DISCUSSION THREAD

**Discussion by BrunoQuaresma** [motiondivision/motion#1651] [Q&A]
Hey folks, I'm trying to use a collapse component inside of a `Reorder.Item` but it is not working as expected. Every time it opens a "weird" animation happens. I think it is related to how the layout animations work but Idk how to solve it, any ideas? Here is a sandbox https://codesandbox.io/s/framer-motion-5-drag-to-reorder-lists-forked-nqzop1?file=/src/Item.tsx

**Comment by paulrostorp** [comment]
Did you find a solution to this @BrunoQuaresma ?

**Reply by hajirasyafi** [reply]
you can try like this https://codesandbox.io/s/framer-motion-5-drag-to-reorder-lists-forked-xluw6i?file=/src/App.tsx

Wrap Reorder.Group inside LayoutGroup :
   ```
<LayoutGroup>
    <Reorder.Group axis="y" onReorder={setItems} values={items}>
      {items.map((item) => (
        <Item key={item} item={item} />
      ))}
    </Reorder.Group>
  </LayoutGroup>
```

then on Reorder.Item add `layout` properties :

`<Reorder.Item layout></Reorder.Item>`

@BrunoQuaresma @paulrostorp

**Comment by abhinav-shram** [comment]
I am facing this issue too. Was anyone able to solve it?
@BrunoQuaresma @paulrostorp @hajirasyafi

**Reply by hajirasyafi** [reply]
do you have sample code? or sandbox? you can try like this one https://codesandbox.io/s/framer-motion-5-drag-to-reorder-lists-forked-xluw6i?file=/src/App.tsx

**Reply by abhinav-shram** [reply]
Yes I tried this link before, it's happening there also did you notice? When the reorder.item component has expandable content inside then it stretches the text of initial/final state weirdly.

**Reply by hajirasyafi** [reply]
Hi @abhinav-shram I think we need add ```AnimatePresence``` inside ```Reorder.Item```

 ```Item.tsx``` component :

```tsx
import { useMotionValue, Reorder, AnimatePresence, motion } from "motion/react";
import { useRaisedShadow } from "./useRaisedShadow";
import { useState } from "react";

interface Props {
  item: string;
}

export const Item = ({ item }: Props) => {
  const y = useMotionValue(0);
  const boxShadow = useRaisedShadow(y);
  const [open, setOpen] = useState(false);

  return (
    <Reorder.Item
      layout="position"
      value={item}
      id={item}
      style={{ boxShadow, y }}
    >
      <div>
        <div>
          <span>{item}</span>
        </div>

        {/* Add Animate Presence */}
        <motion.div className="w-full">
          <AnimatePresence>
            {open && (
              <motion.div
                initial={{ height: 0, opacity: 0, width: "100%" }}
                animate={{ height: "auto", opacity: 1, width: "100%" }}
                exit={{ height: 0, opacity: 0, width: "100%" }}
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
                eiusmod tempor incididunt ut labore et dolore magna aliqua
              </motion.div>
            )}
          </AnimatePresence>
        </motion.div>
        {/* Add Animate Presence */}

        <button
          onClick={() => setOpen((v) => !v)}
          className="px-4 bg-blue-100 rounded"
        >
          Show More
        </button>
      </div>
    </Reorder.Item>
  );
};

```

this is my code
https://codesandbox.io/p/devbox/motion-drag-reorder-mns79x

**Reply by abhinav-shram** [reply]
Hi @hajirasyafi . Thank you for your help. However, I migrated the feature to use dndkit because we wanted more customization than motion offered. I'll make sure to keep this solution in mind when I work with motion components again.
